support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -640,55 +674,132 @@ function CheckoutComponent({ geoId }: Props) {
 						>
 							<Box cssOverrides={shorterBoxMargin}>
 								<BoxContents>
-									<PersonalDetails
-										email={email}
-										firstName={firstName}
-										lastName={lastName}
-										isSignedIn={isSignedIn}
-										// TODO - ONE_OFF support, this should be true when ONE_OFF
-										hideNameFields={false}
-										onEmailChange={(email) => {
-											setEmail(email);
-										}}
-										onFirstNameChange={(firstName) => {
-											setFirstName(firstName);
-										}}
-										onLastNameChange={(lastName) => {
-											setLastName(lastName);
-										}}
-										errors={{}}
-										signOutLink={<Signout isSignedIn={isSignedIn} />}
-										contributionState={
-											showStateSelect && (
-												<StateSelect
-													countryId={countryId}
-													state={'STATE'}
-													onStateChange={() => {
-														//  no-op
+									<div css={personalDetailsFieldGroupStyles(true)}>
+										<h2 css={personalDetailsHeader}>Your details</h2>
+										<div>
+											<TextInput
+												id="email"
+												data-qm-masking="blocklist"
+												label="Email address"
+												value={email}
+												type="email"
+												autoComplete="email"
+												onChange={(event) => setEmail(event.target.value)}
+												disabled={isSignedIn}
+												name="email"
+												required
+												error={emailError}
+												onInvalid={(event) => {
+													preventDefaultValidityMessage(event.currentTarget);
+													const validityState = event.currentTarget.validity;
+													if (validityState.valid) {
+														setEmailError(undefined);
+													} else {
+														if (validityState.valueMissing) {
+															setEmailError('Please enter your email address.');
+														} else {
+															setEmailError(
+																'Please enter a valid email address.',
+															);
+														}
+													}
+												}}
+											/>
+										</div>
+
+										<Signout isSignedIn={isSignedIn} />
+
+										<>
+											<div>
+												<TextInput
+													id="firstName"
+													data-qm-masking="blocklist"
+													label="First name"
+													value={firstName}
+													autoComplete="given-name"
+													autoCapitalize="words"
+													onChange={(event) => setFirstName(event.target.value)}
+													name="firstName"
+													required
```

Comment:
Nice to see the built in HTML attributes being used for validation. Have we checked the rules currently applied to the equivalent form fields on the original checkout are covered here? For example, I think there are max length limits on some of these fields (as implemented here https://github.com/guardian/support-frontend/pull/4168)

Reply:
I've added a [`maxLength` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength) to the input fields.

This doesn't error however, but rather stop a person adding more input. I'm not sure this a great UX as per [some a11y docs](https://a11ysupport.io/tech/html/maxlength_attribute#:~:text=The%20attribute%20prevents%20characters%20beyond,they%20are%20in%20fact%20not.)

> The attribute prevents characters beyond the maximum length from being entered. The user needs to know when the maximum length has been reached, so that they do not think that they are entering data when they are in fact not.

I'm not sure our UX is great either, where we allow you to go on, but we know we're going to block you anyway - wasting your time.

I'll see if I can get to "better" by responding to maxlength being reached.

Reply:
@GHaberis Why can't we have emoji?

Reply:
> @GHaberis Why can't we have emoji?

It'll be an issue for a downstream system such as Salesforce, Braze or Zuora. I'm afraid I can't recall which, maybe the PR history helps.

Reply:
Looks like it was Zuora https://github.com/guardian/support-frontend/pull/3371

Reply:
Interesting.

I've added this as a [standard `pattern` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern) for now - but we might have to see how it scales.

I think it should be alright as most of our [current form validations](https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/forms/formValidation.ts) are covered by the `maxLength`, `required` and a simple `pattern`.